### PR TITLE
#7 setup of db

### LINKED ISSUE #7: Set Up DB
* Create the DB / table from the excel sheet automatically 
* Connect Flask Sql alchemy with the flask app

### PR DESCRIPTION
**Pull request by elainoz**
database created, have yet to connect to flask

### CODE REVIEW

File: setupDB.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
+from flask import Flask
+from flask_sqlalchemy import SQLAlchemy
+from sqlalchemy import Column, Integer, Float, Date, String, VARCHAR
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy import create_engine
+from sqlalchemy.orm import sessionmaker
+import csv
+import pandas as pd
+
+# app = Flask(__name__)
```
Comment (g1isgone) [inline setupDB.py:8]: Let's remove comments in the next commit. Since this file will have to be merged with main.py where the flaskapp is instantiated, it doesn't matter for now  :)

File: setupDB.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+# app.config['SQLALCHEMY_DATABASE_URI'] = 'sqlite:////tmp/test.db'
+# db = SQLAlchemy(app)
+
+Base = declarative_base()
+
+
+class Conferencedb(Base):
```
Comment (g1isgone) [inline setupDB.py]: let's make this ConferenceDB

File: setupDB.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    latinX = Column(VARCHAR)
+    firstgen = Column(VARCHAR)
+
+if __name__ == '__main__':
+    engine = create_engine('sqlite:////tmp/test.db')
+    Base.metadata.create_all(engine)
+    file_name = 'womenInTechConferences.csv'
+    df = pd.read_csv(file_name)
+    df.to_sql(name = Conferencedb.__tablename__, con=engine, index_label='id', if_exists='replace')
+    print(engine.execute("SELECT * FROM womenInTechResources").fetchall())
```
Comment (g1isgone) [inline setupDB.py:43]: let's add in the README.md how to run this db - it can be a separate commit or task

File: setupDB.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+# app = Flask(__name__)
+# app.config['SQLALCHEMY_DATABASE_URI'] = 'sqlite:////tmp/test.db'
+# db = SQLAlchemy(app)
+
+Base = declarative_base()
+
+
+class Conferencedb(Base):
+    __tablename__ = 'womenInTechResources'
+    __table_args__ = {'sqlite_autoincrement': True}
```
Comment (g1isgone) [inline setupDB.py]: out of curiosity what is the settings for ? :)

**Review by g1isgone — APPROVED** [review]

**Review by g1isgone — APPROVED** [review]

**Comment by g1isgone** [comment]
Nice work @kvg-png @elainoz & @ntresvalles1!

File: setupDB.py (hunk 4 above)
Comment (elainoz) [reply setupDB.py]: Line 19 was added to avoid sqlite reusing id's that might have been deleted. Does this case apply to our database? (can be removed if unnecessary)

File: setupDB.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+Base = declarative_base()
+
+
+class ConferenceDB(Base):
+    __tablename__ = 'womenInTechResources'
+    #__table_args__ = {'sqlite_autoincrement': True}
+
+    id = Column(Integer, primary_key=True, nullable=False)
+    name = Column(VARCHAR(40))
```
Comment (lillianjiang) [inline setupDB.py:20]: For `name=Column(VARCHAR(40))` , what does 40 stands for? how is it different from `Column(VARCHAR)` in below? I am new to db so just out of curiosity:)

Comment (smithiesincstech) [reply setupDB.py:20]: believe it tells you the max length of characters (in this case 40) you can store in that column

File: setupDB.py (hunk 4 above)
Comment (smithiesincstech) [reply setupDB.py]: ahh I see okay - in that case we can just leave it in there :)
